stride/physics/iso_elastic/devito.py

```diff
@@ -0,0 +1,398 @@
+        self._max_wavelet = 0.
+        self._src_scale = 0.
+        self._bandwidth = 0.
+
+        self.dev_grid = GridDevito(self.space_order, self.time_order, **kwargs)
+
+        kwargs.pop('grid', None)
+        self.state_operator = OperatorDevito(self.space_order, self.time_order,
+                                             name='elastic_iso_state',
+                                             grid=self.dev_grid,
+                                             **kwargs)
+        self.adjoint_operator = OperatorDevito(self.space_order, self.time_order,
+                                               name='elastic_iso_adjoint',
+                                               grid=self.dev_grid,
+                                               **kwargs)
+        self.boundary = None
+
+    def clear_operators(self):
+        self.state_operator.devito_operator = None
+        self.adjoint_operator.devito_operator = None
+
+    # forward
+
+    async def before_forward(self, wavelets, vp, vs, rho, **kwargs):
+        """
+        Prepare the problem type to run the state or forward problem.
+
+        Parameters
+        ----------
+        wavelets : Traces
+            Source wavelets.
+        vp : ScalarField
+            Compressional (acoustic) speed of sound of the medium, in [m/s].
+        vs : ScalarField
+            Transverse (shear) speed of sound of the medium, in [m/s].
+        rho : ScalarField
+            Density of the medium in [kg/m^3].
+        problem : Problem
+            Sub-problem being solved by the PDE.
+        save_wavefield : bool or int, optional
+            Whether or not to solve the forward wavefield, defaults to True when
+            a gradient is expected, and to False otherwise. An integer number N can
+            also be provided, in which case the last N timesteps of the wavefield
+            will be saved.
+        boundary_type : str, optional
+            Type of boundary for the wave equation (``sponge_boundary_2`` or
+            ``complex_frequency_shift_PML_2``), defaults to ``sponge_boundary_2``.
+        interpolation_type : str, optional
+            Type of source/receiver interpolation (``linear`` or ``hicks``), defaults
+            to ``linear``.
+        attenuation_power : int, optional
+            Power of the attenuation law if attenuation is given (``0`` or ``2``),
+            defaults to ``0``.
+        drp : bool, optional
+            Whether or not to use dispersion-relation preserving coefficients (only
+            available in some versions of Stride). Defaults to False.
+        kernel : str, optional
+            Type of time kernel to use (``OT2`` for 2nd order in time or ``OT4`` for 4th
+            order in time). If not given, it is automatically decided given the time spacing.
+        undersampling : int, optional
+            Amount of undersampling in time when saving the forward wavefield. If not given,
+            it is calculated given the bandwidth.
+
```

Comment:
same thing happens with all these parameters, you should check all documentation

Reply:
Correction made